partiql-tests-data/fail/static-analysis/primitives/time-constructor.ion

```diff
@@ -53,3 +53,172 @@
         result: StaticAnalysisFail
     },
 }
+
+{
+    name: "invalid TIME with STRING literal",
+    statement: "TIME 'time_string'",
+    assert: {
+        result: StaticAnalysisFail
+    },
+}
+
+{
+    name: "invalid TIME with DATE string",
+    statement: "TIME '2012-12-12'",
+    assert: {
+        result: StaticAnalysisFail
+    },
+}
+
+{
+    name: "invalid TIME with just the hour",
+    statement: "TIME '12'",
+    assert: {
+        result: StaticAnalysisFail
+    },
+}
+
+{
+    name: "invalid TIME with just the hour and minute",
+    statement: "TIME '12:30'",
+    assert: {
+        result: StaticAnalysisFail
+    },
+}
+
+{
+    name: "invalid TIME with nano missing a component",
+    statement: "TIME '59.12345'",
+    assert: {
+        result: StaticAnalysisFail
+    },
+}
+
+{
+    name: "invalid TIME without padding of hour",
+    statement: "TIME '1:30:38'",
+    assert: {
+        result: StaticAnalysisFail
+    },
+}
+
+{
+    name: "invalid TIME too many properties",
+    statement: "TIME '12.123:45.123:54.123'",
+    assert: {
+        result: StaticAnalysisFail
+    },
+}
+
+{
+    name: "invalid TIME minus before hours",
+    statement: "TIME '-19:45:13'",
+    assert: {
+        result: StaticAnalysisFail
+    },
+}
+
+{
+    name: "invalid TIME missing plus or minus before offset",
+    statement: "TIME '23:59:59.99999 05:30'",
+    assert: {
+        result: StaticAnalysisFail
+    },
+}
+
+{
+    name: "invalid TIME offset with DECIMAL",
+    statement: "TIME '23:59:59+05:30.00'",
+    assert: {
+        result: StaticAnalysisFail
+    },
+}
+
+{
+    name: "invalid TIME with unmatched left paren",
+    statement: "TIME ( '23:59:59.99999'",
+    assert: {
+        result: StaticAnalysisFail
+    },
+}
+
+{
+    name: "invalid TIME with no precision value",
+    statement: "TIME () '23:59:59.99999'",
+    assert: {
+        result: StaticAnalysisFail
+    },
+}
+
+{
+    name: "invalid TIME precision in square brackets",
+    statement: "TIME [4] '23:59:59.99999'",
+    assert: {
+        result: StaticAnalysisFail
+    },
+}
+
+{
+    name: "invalid TIME precision in curly braces",
+    statement: "TIME {4} '23:59:59.99999'",
+    assert: {
+        result: StaticAnalysisFail
+    },
+}
+
+{
+    name: "invalid TIME precision not in parens",
+    statement: "TIME 4 '23:59:59.99999'",
+    assert: {
+        result: StaticAnalysisFail
+    },
+}
+
+{
+    name: "invalid TIME precision as string",
+    statement: "TIME ('4') '23:59:59.99999'",
+    assert: {
+        result: StaticAnalysisFail
+    },
+}
+
+{
+    name: "invalid TIME precision as string literal",
+    statement: "TIME ('four') '23:59:59.99999'",
+    assert: {
+        result: StaticAnalysisFail
+    },
+}
```

Comment:
I can generally see how the other tests in the PR are more so static analysis-related errors vs syntax. The date/time string with an invalid value that will be detected during some static analysis pass.

I'm less certain about these tests L146-191. These tests are not using the `TIME` constructor properly (e.g. not using parens around the precision argument). Why are these errors not classified as parse errors? Perhaps it's similar to how the function arguments are being modeled?

Reply:
@alancai98 I agree, but looks like we can move back only a subset—which is `L153-L175`—does that sound reasonable to you? and yes the rest are b/c how the function parameters are modeled.

Reply:
Yeah, that subset sounds reasonable to me.